src/Service.Host/client/src/components/StockPools.js

```diff
@@ -0,0 +1,306 @@
+import React, { useEffect, useState } from 'react';
+import Typography from '@mui/material/Typography';
+import moment from 'moment';
+import Dialog from '@mui/material/Dialog';
+import List from '@mui/material/List';
+import Grid from '@mui/material/Grid2';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import { DataGrid, GridSearchIcon } from '@mui/x-data-grid';
+import { Loading, Search, ErrorCard, SnackbarMessage } from '@linn-it/linn-form-components-library';
+import Button from '@mui/material/Button';
+import Page from './Page';
+import config from '../config';
+import itemTypes from '../itemTypes';
+import usePut from '../hooks/usePut';
+import useInitialise from '../hooks/useInitialise';
+import usePost from '../hooks/usePost';
+import useSearch from '../hooks/useSearch';
+
+function StockPools() {
+    const { isLoading, result: stockPoolResult } = useInitialise(itemTypes.stockPools.url);
+    const { isLoading: accountingCompanyLoading, result: accountingCompany } = useInitialise(
+        itemTypes.accountingCompany.url
+    );
+    const [creating, setCreating] = useState(false);
+    const [stockPools, setStockPools] = useState();
+    const [searchTerm, setSearchTerm] = useState();
+    const [snackbarVisible, setSnackbarVisible] = useState(false);
+
+    const {
+        search: searchStorageLocations,
+        results: storageLocationsSearchResults,
+        loading: storageLocationsSearchLoading,
+        clear: clearStorageLocation
+    } = useSearch(itemTypes.storageLocations.url, 'locationId', 'locationCode', 'description');
+
+    const {
+        send: updateStockPool,
+        isLoading: updateLoading,
+        errorMessage: updateError,
+        putResult: updateResult
+    } = usePut(itemTypes.stockPools.url, true);
+
+    const {
+        send: createStockPool,
+        createStockPoolLoading,
+        errorMessage: createStockPoolError,
+        postResult: createStockPoolResult,
+        clearPostResult: clearCreateStockPool
+    } = usePost(itemTypes.stockPools.url);
+
+    useEffect(() => {
+        setSnackbarVisible(!!updateResult || !!createStockPoolResult);
+    }, [createStockPoolResult, updateResult]);
+
+    useEffect(() => {
+        setStockPools(stockPoolResult);
+    }, [stockPoolResult]);
+
+    const [searchDialogOpen, setSearchDialogOpen] = useState({
+        forRow: null,
+        forColumn: null
```

Comment:
also don't need to track which column the search is open for - there is only one possibility